Formula/libxml2.rb

```diff
@@ -93,8 +102,8 @@ def install
         return 0;
       }
     EOS
-    args = shell_output("#{bin}/xml2-config --cflags --libs").split
-    args += %w[test.c -o test]
+    args = %w[test.c -o test]
+    args += shell_output("#{bin}/xml2-config --cflags --libs").split
```

Comment:
What is the point of this reordering?

Reply:
On ubuntu (and on ubunut only (since ubuntu 6.04 I think), they decided that the flag order for c/c++ was different. Using this new order makes sure it works on mac, and whatever linux is there.

Reply:
I think it has to do with some compiler/linker magic and library import order.

Reply:
Quoting @sjackman :

> Also in Natty, ld runs with the --as-needed option enabled by default. This means that, in the example above, if no symbols from libwheel were needed by racetrack, then libwheel would not be linked even if it was explicitly included in the command-line compiler flags. NOTE: The ld --as-needed default was reverted for the final natty release, and will be re-enabled in the o-series.
https://wiki.ubuntu.com/NattyNarwhal/ToolchainTransition
So Ubuntu has since Ubuntu 11.10 (Oneiric Ocelot) enabled ld --as-needed by default, which causes this difference in default behaviour.